rclcpp/include/rclcpp/memory_strategy.hpp

```diff
@@ -38,137 +36,171 @@ namespace memory_strategy
+  static rclcpp::service::ServiceBase::SharedPtr
+  get_service_by_handle(void * service_handle,
+    const WeakNodeVector & weak_nodes)
   {
-    switch (type) {
-      case HandleType::subscription_handle:
-        if (handles != subscription_handles.data()) {
-          throw std::runtime_error(
-                  "tried to return memory that isn't handled by this MemoryStrategy");
-        }
-        memset(handles, 0, subscription_handles.size());
-        break;
-      case HandleType::service_handle:
-        if (handles != service_handles.data()) {
-          throw std::runtime_error(
-                  "tried to return memory that isn't handled by this MemoryStrategy");
-        }
-        memset(handles, 0, service_handles.size());
-        break;
-      case HandleType::client_handle:
-        if (handles != client_handles.data()) {
-          throw std::runtime_error(
-                  "tried to return memory that isn't handled by this MemoryStrategy");
+    for (auto & weak_node : weak_nodes) {
+      auto node = weak_node.lock();
+      if (!node) {
+        continue;
+      }
+      for (auto & weak_group : node->get_callback_groups()) {
+        auto group = weak_group.lock();
+        if (!group) {
+          continue;
         }
-        memset(handles, 0, client_handles.size());
-        break;
-      case HandleType::guard_condition_handle:
-        if (handles != guard_cond_handles.data()) {
-          throw std::runtime_error(
-                  "tried to return memory that isn't handled by this MemoryStrategy");
+        for (auto & service : group->get_service_ptrs()) {
+          if (service->get_service_handle()->data == service_handle) {
+            return service;
+          }
         }
-        guard_cond_handles.fill(0);
-        break;
-      default:
-        throw std::runtime_error("Unknown HandleType " + std::to_string(static_cast<int>(type)) +
-                ", could not borrow handle memory.");
+      }
     }
+    return rclcpp::service::ServiceBase::SharedPtr();
   }
 
-  /// Provide a newly initialized AnyExecutable object.
-  // \return Shared pointer to the fresh executable.
-  virtual executor::AnyExecutable::SharedPtr instantiate_next_executable()
+  static rclcpp::client::ClientBase::SharedPtr
+  get_client_by_handle(void * client_handle,
+    const WeakNodeVector & weak_nodes)
   {
-    return std::make_shared<executor::AnyExecutable>();
+    for (auto & weak_node : weak_nodes) {
+      auto node = weak_node.lock();
+      if (!node) {
+        continue;
+      }
+      for (auto & weak_group : node->get_callback_groups()) {
+        auto group = weak_group.lock();
+        if (!group) {
+          continue;
+        }
+        for (auto & client : group->get_client_ptrs()) {
+          if (client->get_client_handle()->data == client_handle) {
+            return client;
+          }
+        }
+      }
+    }
+    return rclcpp::client::ClientBase::SharedPtr();
```

Comment:
The same comment applies here as above about returning a `nullptr` instead. I think there are more cases below too.